src/coreclr/src/inc/corinfo.h

```diff
@@ -338,6 +338,139 @@ struct SYSTEMV_AMD64_CORINFO_STRUCT_REG_PASSING_DESCRIPTOR
+    {
+        unsigned baseSize     = sizeof(CORINFO_PATCHPOINT_INFO);
+        unsigned variableSize = localCount * sizeof(int);
+        unsigned totalSize    = baseSize + variableSize;
+        return totalSize;
+    }
+
+    // Initialize
+    void Initialize(unsigned localCount, unsigned ilSize, int fpToSpDelta)
+    {
+        m_patchpointInfoSize      = ComputeSize(localCount);
+        m_ilSize                  = ilSize;
+        m_fpToSpDelta             = fpToSpDelta;
+        m_numberOfLocals          = localCount;
+        m_genericContextArgOffset = -1;
+        m_keptAliveThisOffset     = -1;
+        m_securityCookieOffset    = -1;
+    }
+
+    // Total size of this patchpoint info record, in bytes
+    unsigned PatchpointInfoSize() const
+    {
+        return m_patchpointInfoSize;
+    }
+
+    // IL Size of the original method
+    unsigned ILSize() const
+    {
+        return m_ilSize;
+    }
+
+    // FP to SP delta of the original method
+    int FpToSpDelta() const
+    {
+        return m_fpToSpDelta;
+    }
+
+    // Number of locals in the original method (including special locals)
+    unsigned NumberOfLocals() const
+    {
+        return m_numberOfLocals;
+    }
+
+    // Original method caller SP offset for generic context arg
+    int GenericContextArgOffset() const
+    {
+        return m_genericContextArgOffset;
+    }
+
+    void SetGenericContextArgOffset(int offset)
+    {
+        m_genericContextArgOffset = offset;
+    }
+
+    // Original method FP relative offset for kept-alive this
+    int KeptAliveThisOffset() const
+    {
+        return m_keptAliveThisOffset;
+    }
+
+    bool HasKeptAliveThis() const
+    {
+        return m_keptAliveThisOffset != -1;
+    }
+
+    void SetKeptAliveThisOffset(int offset)
+    {
+        m_keptAliveThisOffset = offset;
+    }
+
+    // Original method FP relative offset for security cookie
+    int SecurityCookieOffset() const
+    {
+        return m_securityCookieOffset;
+    }
+
+    bool HasSecurityCookie() const
+    {
+        return m_securityCookieOffset != -1;
+    }
+
+    void SetSecurityCookieOffset(int offset)
+    {
+        m_securityCookieOffset = offset;
+    }
+
+    // True if this local was address exposed in the original method
+    bool IsExposed(unsigned localNum) const
+    {
+        return ((m_offsetAndExposureData[localNum] & EXPOSURE_MASK) != 0);
+    }
+
+    void SetIsExposed(unsigned localNum)
+    {
+        m_offsetAndExposureData[localNum] |= EXPOSURE_MASK;
+    }
+
+    // FP relative offset of this local in the original method
+    int Offset(unsigned localNum) const
+    {
+        return (m_offsetAndExposureData[localNum] & ~EXPOSURE_MASK);
+    }
+
+    void SetOffset(unsigned localNum, int offset)
+    {
+        m_offsetAndExposureData[localNum] = offset;
+    }
+
+private:
+    enum
+    {
+        EXPOSURE_MASK = 0x1
+    };
+
+    unsigned m_patchpointInfoSize;
+    unsigned m_ilSize;
```

Comment:
This is kept around just for debug-only check. Should be under `#ifdef DEBUG`

Reply:
Will be tricky to make this conditional as it will break mixing and match jit and runtime build flavors.

Reply:
So just delete it? It does not seem to be a high-value check.

Reply:
Yes, that's what I was planning to do.

Reply:
Done.